src/receiver.ts

```diff
@@ -113,33 +114,43 @@ export default class Receiver extends EventEmitter {
-                            response.url,
-                            response.headers.get('content-type') || "application/ld+json"
-                        )
-
-                        // transform bodystream
-                        //const bodyStream = new ReadableWebToNodeStream(response.body || new ReadableStream())
-
-                        // TODO: Fix this when NodeJS vs. Stream API chaos is over
-                        const bodyStream = new Readable()
-                        bodyStream.push(await response.text())
-                        bodyStream.push(null)
-
-                        // parse the notification
-                        const notification = await EventNotification.parse(bodyStream, jsonldParser)
-
-                        // emit an event with notification
-                        const idToCheck = strategy == 'notification_id' ? item.url : notification.id.value
-
-                        if (this.db && !this.db.get(idToCheck)) {
-                            this.emit('notification', notification)
-                            this.db.set(idToCheck, true)
+                        try {
+                            // parse the notification
+                            const jsonldParser = JsonLdParser.fromHttpResponse(
+                                response.url,
+                                response.headers.get('content-type') || "application/ld+json"
+                            )
+
+                            // transform bodystream
+                            //const bodyStream = new ReadableWebToNodeStream(response.body || new ReadableStream())
+
+                            // TODO: Fix this when NodeJS vs. Stream API chaos is over
+                            const bodyStream = new Readable()
+                            bodyStream.push(responseText)
+                            bodyStream.push(null)
+
+                            // parse the notification
+                            const notification = await EventNotification.parse(bodyStream, jsonldParser)
+
+                            // emit an event with notification
+                            const idToCheck = strategy == 'notification_id' ? item.url : notification.id.value
+
+                            if (this.db && !this.db.get(idToCheck)) {
+                                this.emit('notification', notification)
+                                this.db.set(idToCheck, true)
+                            }
+                        }
+                        catch (e) {
+                            const idToCheck = md5.default(responseText)
+                            if (this.db && !this.db.get(idToCheck)) {
+                                this.emit('ignore',idToCheck)
+                                this.db.set(idToCheck, true)
+                            }
                         }
                     } catch (e) {
-                        this.emit('error', e)
+                        this.emit('network_error', e)
```

Comment:
Why did the error message need to change? I would prefer no breaking changes because I have code already depending on this.